spec/classes/foreman_proxy_content__container_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+describe 'foreman_proxy_content::container' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+
+      describe 'with inherited parameters' do
+
+        let(:pre_condition) do
+          <<-PUPPET
+          class { 'foreman_proxy':
+            foreman_ssl_cert => "/etc/foreman-proxy/foreman_ssl_cert.pem",
+            foreman_ssl_key => "/etc/foreman-proxy/foreman_ssl_key.pem",
+          }
+          include foreman_proxy_content
+          PUPPET
+        end
+
+        it { is_expected.to compile.with_all_deps }
+        it do
+          is_expected.to contain_apache__vhost__fragment('pulp-https-container')
+            .with_vhost('pulpcore-https')
+            .with_priority('10')
+          # TODO: test http_content
```

Comment:
Trying to figure out how I can test the `vhost` itself. Perhaps there's a way to validate `$https_content` ?

Reply:
Answered this myself: `.with_content(...)`